src/main/webapp/app/lecture/lecture-unit/lecture-unit-management/attachment-units/attachment-units.component.ts

```diff
@@ -0,0 +1,181 @@
+    ) {
+        this.file = this.router.getCurrentNavigation()!.extras.state!.file;
+        this.fileName = this.router.getCurrentNavigation()!.extras.state!.fileName;
+        const lectureRoute = this.activatedRoute.parent!.parent!;
+        combineLatest([lectureRoute.paramMap, lectureRoute.parent!.paramMap]).subscribe(([params]) => {
+            this.lectureId = Number(params.get('lectureId'));
+            this.courseId = Number(params.get('courseId'));
+        });
+    }
+
+    ngOnInit(): void {
+        this.isLoading = true;
+        this.isProcessingMode = true;
+
+        const formData: FormData = new FormData();
+        formData.append('file', this.file);
+
+        this.attachmentUnitService.getSplitUnitsData(this.lectureId, formData).subscribe({
+            next: (res: any) => {
+                if (res) {
+                    this.units = res.body.lectureUnitDTOS;
+                    this.numberOfPages = res.body.numberOfPages;
+                    this.isLoading = false;
+                }
+            },
+            error: (res: HttpErrorResponse) => {
+                if (res.error.params === 'file' && res?.error?.title) {
+                    this.alertService.error(res.error.title);
+                } else {
+                    onError(this.alertService, res);
+                }
+                this.isLoading = false;
+            },
+        });
+    }
+
+    createAttachmentUnits(): void {
+        this.isLoading = true;
+
+        const formData: FormData = new FormData();
+        formData.append('file', this.file);
+        formData.append('lectureUnitSplitDTOs', objectToJsonBlob(this.units));
+
+        this.attachmentUnitService.createUnits(this.lectureId, formData).subscribe({
+            next: () => {
+                this.router.navigate(['../../'], { relativeTo: this.activatedRoute });
+                this.isLoading = false;
+            },
+            error: (res: HttpErrorResponse) => {
+                if (res.error.params === 'file' && res?.error?.title) {
+                    this.alertService.error(res.error.title);
+                } else {
+                    onError(this.alertService, res);
+                }
+                this.isLoading = false;
+            },
+        });
+    }
+
+    cancelSplit() {
+        this.router.navigate(['course-management', this.courseId.toString(), 'lectures', this.lectureId.toString()]);
+    }
+
+    addRow() {
+        const unitDynamic = {
+            unitName: '',
+            startPage: 0,
+            endPage: 0,
+        };
+        this.units.push(unitDynamic);
+        return true;
+    }
+
+    deleteRow(i: number) {
+        if (this.units.length === 1) {
+            return false;
+        } else {
+            this.units.splice(i, 1);
+            return true;
+        }
+    }
+
+    validUnitInformation(): boolean {
+        for (const unit of this.units) {
+            if (unit.unitName === '' || unit.unitName === undefined) {
```

Comment:
```suggestion
            if (!unit.unitName) {
```
Same below. Also why is an empty string sometimes allowed and sometimes not?

Reply:
When it's allowed and for which fields?